### PR TITLE
CASSANDRA-17239 for cassandra-3.0

### PR DESCRIPTION
**Pull request by jonmeredith**
See [CASSANDRA-17239](https://issues.apache.org/jira/browse/CASSANDRA-17239)

### CODE REVIEW

File: test/unit/org/apache/cassandra/db/compaction/CompactionExecutorTest.java [hunk 1]
```diff
@@ -63,44 +70,43 @@ public void destroy() throws Exception
         executor.awaitTermination(1, TimeUnit.MINUTES);
     }
 
+    void awaitExecution() throws Exception
+    {
+        assert afterExecuteCompleted.await(10, TimeUnit.SECONDS) : "afterExecute failed to complete";
```
Comment (jmckenzie-dev) [inline test/unit/org/apache/cassandra/db/compaction/CompactionExecutorTest.java:75]: You inherited this from the initial implementation, but is there a reason we're only allowing for 10 seconds to execute before timing out? This seems like the kind of thing that could flake in adverse environments / scheduling / k8s silliness in the future.

**Review by jmckenzie-dev — APPROVED** [review]
One question about the timeout selected (and by that I mean inherited) but otherwise lgtm.

File: test/unit/org/apache/cassandra/db/compaction/CompactionExecutorTest.java (hunk 1 above)
Comment (jonmeredith) [reply test/unit/org/apache/cassandra/db/compaction/CompactionExecutorTest.java:75]: Happy to bump it up to a minute - the original test doesn't have a timeout as far as I can tell - it's just sleeping 10 milliseconds between checks on the future.

Comment (jmckenzie-dev) [reply test/unit/org/apache/cassandra/db/compaction/CompactionExecutorTest.java:75]: You're absolutely right. I was thinking it was sleeping 10 seconds and that's why you moved to a 10 second timeout on the signal wait; totally spaced that sleep's ms. 👍

**Pull request closed without merging**